### PR TITLE
feat: modal support cssVar

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | modal support css variable theme |
| 🇨🇳 Chinese | 弹框支持 css 变量主题 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 4481e5a</samp>

This pull request refactors the modal component style to use custom hooks and CSS variables for dynamic theming and customization. It uses the `@ant-design/cssinjs` package to simplify the style calculations and the CSS variable generation. It also adds server-side rendering support for the CSS variables. The main files affected are `components/modal/Modal.tsx`, `components/modal/PurePanel.tsx`, `components/modal/style/confirmCmp.tsx`, `components/modal/style/index.tsx`, and `components/modal/style/cssVar.tsx`.

### <samp>🤖 Generated by Copilot at 4481e5a</samp>

*  Register and apply CSS variables for modal component using custom hooks ([link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a3edcd43b644d0539a67f65e68ba30d6acb77a12531a6a71d1e26a008a9d752cR19-R20), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a3edcd43b644d0539a67f65e68ba30d6acb77a12531a6a71d1e26a008a9d752cL96-R100), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a3edcd43b644d0539a67f65e68ba30d6acb77a12531a6a71d1e26a008a9d752cL122-R126), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a3edcd43b644d0539a67f65e68ba30d6acb77a12531a6a71d1e26a008a9d752cL132-R136), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27R14), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27R39), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27L69-R71), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-9103d37c6668043c170813d54cb43ca28bc6cf111eedfc6718aeae7f540a0b27L83-R85), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-6590133ad83927e5fecbd118230bccc3158a134697c42429c0ceac9705f58eb7R1-R8))
*  Import and use `unit` function from `@ant-design/cssinjs` package to convert numeric values to CSS unit strings ([link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-99ef0451fefa19a2e28ee86c53122dd8bdde5704ca454859e5e7230c05df45dfR2), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486R9), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L156-R169), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L175-R188), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L196-R220), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L222-R239), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L244-R261), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L307-R326), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L316-R341))
*  Replace arithmetic expressions with `token.calc` and `unit` methods to perform calculations with token values and CSS variables ([link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-99ef0451fefa19a2e28ee86c53122dd8bdde5704ca454859e5e7230c05df45dfL44-R47), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-99ef0451fefa19a2e28ee86c53122dd8bdde5704ca454859e5e7230c05df45dfL56-R58), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L325-R350), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L360-R393), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L372-R402))
*  Add `titleHeight` and `contentHeight` properties to `ModalToken` interface and token values to store and compute the height of modal title and content elements ([link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-99ef0451fefa19a2e28ee86c53122dd8bdde5704ca454859e5e7230c05df45dfL16-R21), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L60-R74), [link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486R415-R416))
*  Change the type of some token properties from `number` to `number | string` to allow different CSS units ([link](https://github.com/ant-design/ant-design/pull/45798/files?diff=unified&w=0#diff-a27b99943336f4a3b470158edbccad703a421adf1b1453c0c006e0794c7db486L47-R48))

### CODE REVIEW

File: components/modal/style/index.tsx [hunk 1]
```diff
@@ -357,20 +382,25 @@ export const prepareToken: (token: Parameters<GenStyleFn<'Modal'>>[0]) => ModalT
...
     modalFooterBorderStyle: token.lineType,
     modalFooterPaddingVertical: token.paddingXS,
     modalFooterPaddingHorizontal: token.padding,
     modalFooterBorderWidth: token.lineWidth,
     modalIconHoverColor: token.colorIconHover,
     modalCloseIconColor: token.colorIcon,
-    modalCloseBtnSize: token.fontSize * token.lineHeight,
-    modalConfirmIconSize: token.fontSize * token.lineHeight,
+    modalCloseBtnSize: token.calc(token.fontSize).mul(token.lineHeight).equal(),
+    modalConfirmIconSize: token.calc(token.fontSize).mul(token.lineHeight).equal(),
```
Comment (MadCcc) [inline components/modal/style/index.tsx:401]: 这两个用 fontHeight 替换下

File: components/modal/style/index.tsx [hunk 2]
```diff
@@ -57,8 +58,20 @@ export interface ModalToken extends FullToken<'Modal'> {
...
+  /**
+   * @desc 标题高度
+   * @descEN Height of title
+   */
+  titleHeight: number | string;
+  /**
+   * @desc 内容区域高度
+   * @descEN Height of content
+   */
+  contentHeight: number | string;
```
Comment (MadCcc) [inline components/modal/style/index.tsx:74]: 非 component token 加上 modal 前缀

Comment (kiner-tang) [reply components/modal/style/index.tsx]: done

File: components/modal/style/index.tsx (hunk 1 above)
Comment (kiner-tang) [reply components/modal/style/index.tsx]: done

File: components/modal/style/index.tsx [hunk 3]
```diff
@@ -357,20 +372,25 @@ export const prepareToken: (token: Parameters<GenStyleFn<'Modal'>>[0]) => ModalT
...
     modalFooterPaddingHorizontal: token.padding,
     modalFooterBorderWidth: token.lineWidth,
     modalIconHoverColor: token.colorIconHover,
     modalCloseIconColor: token.colorIcon,
-    modalCloseBtnSize: token.fontSize * token.lineHeight,
-    modalConfirmIconSize: token.fontSize * token.lineHeight,
+    modalCloseBtnSize: token.fontHeight,
+    modalConfirmIconSize: token.fontHeight,
+    modalTitleHeight: token.calc(token.titleFontSize).mul(token.titleLineHeight).equal(),
+    modalContentHeight: token.calc(token.fontSize).mul(token.lineHeight).equal(),
```
Comment (MadCcc) [inline components/modal/style/index.tsx:393]: 这个也可以直接用 fontHeight，不用加了

**Review by MadCcc — APPROVED** [review]

**Pull request merged**